pkg/aws/g8scontrolplane/admit_g8scontrolplane_infraref_test.go

```
@@ -0,0 +1,135 @@
+				if err != nil {
+					t.Fatal(err)
+				}
+			}
+
+			// run admission request for g8sControlPlane without reference
+			var patch []admission.PatchOperation
+			patch, err = admit.Admit(g8sControlPlaneNoReferenceAdmissionRequest())
+			if err != nil {
+				t.Fatal(err)
+			}
+			// parse patches
+			for _, p := range patch {
+				if p.Path == "/spec/infrastructureRef" {
+					updatedInfraRefName = p.Value.(*v1.ObjectReference).Name
+				}
+			}
+			// check if the reference patch is as expected
+			if tc.expectedReferenceName != updatedInfraRefName {
+				t.Fatalf("expected %#q to be equal to %#q", tc.expectedReferenceName, updatedInfraRefName)
+			}
+		})
+	}
+}
+
+func g8sControlPlaneNoReferenceAdmissionRequest() *v1beta1.AdmissionRequest {
+	req := &v1beta1.AdmissionRequest{
+		Kind: metav1.GroupVersionKind{
+			Version: "infrastructure.giantswarm.io/v1alpha2",
+			Kind:    "G8sControlPlane",
+		},
+		Resource: metav1.GroupVersionResource{
+			Version:  "infrastructure.giantswarm.io/v1alpha2",
+			Resource: "g8scontrolplanes",
+		},
+		Operation: v1beta1.Create,
+		Object: runtime.RawExtension{
+			Raw:    getG8sControlPlaneNoRefRAWByte(),
+			Object: nil,
+		},
+	}
+	return req
+}
+
+func getG8sControlPlaneNoRefRAWByte() []byte {
+	g8scontrolPlane := &infrastructurev1alpha2.G8sControlPlane{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "G8sControlPlane",
+			APIVersion: "infrastructure.giantswarm.io/v1alpha2",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      controlPlaneName,
+			Namespace: controlPlaneNameSpace,
+			Labels: map[string]string{
+				"giantswarm.io/control-plane":   controlPlaneName,
+				"giantswarm.io/organization":    "giantswarm",
+				"release.giantswarm.io/version": "11.5.0",
+			},
+		},
+		Spec: infrastructurev1alpha2.G8sControlPlaneSpec{
+			Replicas:          1,
+			InfrastructureRef: v1.ObjectReference{},
+		},
+	}
+	byt, _ := json.Marshal(g8scontrolPlane)
```

We should never ignore errors. If you want to have a nice interface and errors are highly unlikely we should panic anyway, just to know that something is off, and where. Imagine you work with a system and it does not work and does not say anything about its own malfunction.